mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala

```diff
@@ -49,7 +49,48 @@ class MultilayerPerceptronClassifierSuite extends SparkFunSuite with MLlibTestSp
     }
   }
 
-  // TODO: implement a more rigorous test
+  test("Test set weights by training restart") {
```

Comment:
I don't quite understand this unit test. It seems that the difference is whether we want to keep using the same random number generator or reset the seed to `12L` every 10 iterations. The solution could converge to a bad local minimal but restarting doesn't always guarantee to fix it.

The content of the test doesn't match the test name. If we want to test setting initial weights, we can show that with proper warm starts, the algorithm converges faster in the same number of iterations.